synphot/wrapper.py

```diff
@@ -0,0 +1,153 @@
+from astropy.modeling.models import Const1D
+
+# LOCAL
+from . import binning, exceptions, units, utils
+from .models import Empirical1D
+from .spectrum import BaseSourceSpectrum, SourceSpectrum, SpectralElement
+from .observation import Observation, ccd_snr, exptime_from_ccd_snr
+
+
+class Observe:
+    """
+    Parameters
+    ----------
+    source_spec : `~synphot.spectrum.SourceSpectrum`
+        The spectrum of the target object. 
+        If distance is given, the spectrum is assumed to be at the source. 
+    distance : `~astropy.units.quantity.Quantity`
+        Distance to the source spectrum. Default is None.
+        If None, source_spec is assumed to be scaled by the distance.
+        If distance is given, the spectrum is assumed to be at the source.
+    diameter : `~astropy.units.quantity.Quantity`
+        Diameter of the telescope aperture.
+    bandpass : str, list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values, or 
+        `~synphot.spectrum.SpectralElement`
+        The bandpass(es) of this observation. Can either be specified by a 
+        string or a list with dimensions 2 x n, where one row is the
+        list of wavelengths associated with the bandpass, and the other
+        row contains the fractional values of the bandpass.
+        Default is a constant model with amplitude = 1, i.e. as if there
+        were no bandpass in effect.
+    quantum_efficiency : list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values
+        A model for the quantum efficiency of the instrument's CCD. 
+        Default is None.
+    atmospheric_transmission : list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values
+        A model for the atmopsheric transmission.
+    """
+    @u.quantity_input(diameter=u.m, distance=u.pc, exptime=u.s)
+    def __init__(self, source_spec, diameter, distance=0 * u.pc,
+                 bandpass=SpectralElement(Const1D, amplitude=1),
+                 quantum_efficiency=None, atmospheric_transmission=None,
+                 force=None, exptime=1 * u.s):
+        self.source_spec = source_spec
+        self.diameter = diameter
+        self.aperture_area = np.pi * (diameter / 2) ** 2
+        self.distance = distance
+        self._bandpass = bandpass
+        self._qe = quantum_efficiency
+        self._at = atmospheric_transmission
+        self.exptime = exptime
+
+        # turn spectrum elements into synphot objects
+        self._set_bandpass()
+        if self._qe is not None:
+            self._include_qe()
+        if self._at is not None:
+            self._include_atmosphere()
+                    
+        self.obs = Observation(self.source_spec, self.throughput, force=force)
+        self.countrate = self.obs.countrate(area=self.aperture_area)
+        self.snr = ccd_snr(self.countrate * self.exptime)
+
+    def _set_bandpass(self):
```

Comment:
You don't strictly need to write docstrings for private functions, and you certainly don't need to write them with the same rigor as you do for public functions, but it's still nice to have a one-line summary for each method, even the private ones. This happens all over the place in astropy, [for example](https://github.com/astropy/astropy/blob/master/astropy/coordinates/sky_coordinate.py#L302-L326).